src/store/builder/index.ts

```diff
@@ -113,27 +117,113 @@ export const mutations: MutationTree<IBuilderState> = {
     operations[operation.kind] = operation
   },
 
-  setBlockPositionTo(state, { position: { x, y }, block }) {
-    // todo: ensure our vendor_metadata.io_viamo is always instantiated with builder // uiData props
-    // if (!block.vendor_metadata.io_viamo.uiData) {
-    //   defaultsDeep(block, {vendor_metadata: {io_viamo: {uiData: {xPosition: 0, yPosition: 0}}}})
-    //   Vue.observable(block.vendor_metadata.io_viamo.uiData)
-    // }
-
-    block.vendor_metadata.io_viamo.uiData.xPosition = x
-    block.vendor_metadata.io_viamo.uiData.yPosition = y
-  },
-
   setIsEditable(state, value) {
     state.isEditable = value
   },
 
   initDraggableForExitsByUuid(state) {
     state.draggableForExitsByUuid = {}
+  },
+
+  updateBlockDraggablePosition(state, { uuid, position }: { uuid: IBlock['uuid'], position: IPosition }) {
+    Object.assign(state.draggableForBlocksByUuid[uuid], { left: position.x, top: position.y })
+  },
+
+  updateToolBarHeight(state, height) {
+    state.toolbarHeight = height
   }
 }
 
 export const actions: ActionTree<IBuilderState, IRootState> = {
+  setBlockPositionTo({ state, commit, getters, dispatch, rootState, rootGetters }, { position: { x, y }, block }) {
+    // todo: ensure our vendor_metadata.io_viamo is always instantiated with builder // uiData props
+    // if (!block.vendor_metadata.io_viamo.uiData) {
+    //   defaultsDeep(block, {vendor_metadata: {io_viamo: {uiData: {xPosition: 0, yPosition: 0}}}})
+    //   Vue.observable(block.vendor_metadata.io_viamo.uiData)
+    // }
+
+    if (!includes(rootState.flow.selectedBlockUuids, block.uuid)) {
+      block.vendor_metadata.io_viamo.uiData.xPosition = x
+      block.vendor_metadata.io_viamo.uiData.yPosition = y
+    } else { // the block is selected
+      // Store UI Position
+      const initialPosition = {
+        x: clone(block.vendor_metadata.io_viamo.uiData.xPosition),
+        y: clone(block.vendor_metadata.io_viamo.uiData.yPosition),
+      } as IPosition
+
+      // Prepare translation
+      const translationDelta: IPosition = {
+        x: x - initialPosition.x,
+        y: y - initialPosition.y
+      }
+
+      let shouldReversePosition = false;
+      if (translationDelta.x > 0 /* moving to the right */ ||
+        rootGetters['flow/selectedBlockAtTheFurthestLeftPosition'].vendor_metadata.io_viamo.uiData.xPosition > 0 /* left space still available */) {
```

Comment:
Since the new getters like `selectedBlockAtTheFurthestLeftPosition` are only used here, can we move them into this file as a modue-level functions that we call directly? This would mitigate iterating every block on the canvas on every change if we're only ever using here :)